lib/rules/selector-type-no-unknown/index.mjs

```diff
@@ -79,13 +84,13 @@ const rule = (primary, secondaryOptions) => {
 					return;
 				}
 
-				if (optionsMatches(secondaryOptions, 'ignoreNamespaces', tagNode.namespace)) {
-					return;
-				}
+				if (optionsMatches(secondaryOptions, 'ignoreNamespaces', tagNode.namespace)) return;
 
-				if (optionsMatches(secondaryOptions, 'ignoreTypes', tagNode.value)) {
-					return;
-				}
+				if (optionsMatches(secondaryOptions, 'ignoreTypes', tagNode.value)) return;
+
+				const value = tagNode.parent?.parent?.value?.toLowerCase() ?? '';
+
+				if (IGNORED_PSEUDO_ELEMENTS.has(value)) return;
```

Comment:
[suggest] I suggest the two points:

1. Extract a helper function in this file for readability. E.g.
   ```suggestion
   				if (isWithinIgnoredPseudoElement(tagNode)) return;
   ```
2. Use the check function of `postcss-selector-parser` for AST safety. E.g.
   ```js
   import selectorParser from 'postcss-selector-parser';
   // ...
   const pseudoElement = tagNode?.parent?.parent;
   if (selectorParser.isPseudoElement(pseudoElement) && IGNORED_PSEUDO_ELEMENTS.has(pseudoElement.value)) return;
   ```